external-config/src/main/groovy/grails/plugin/externalconfig/ExternalConfigRunListener.groovy

```diff
@@ -138,15 +138,15 @@ class ExternalConfigRunListener implements SpringApplicationRunListener {
         WriteFilteringMap filterMap = new WriteFilteringMap(currentConfig)
         slurper.binding = filterMap
         ConfigObject configObject = slurper.parse(configText)
-        Map properties = configText ? configObject?.flatten() : [:]
+        Map<String, Object> properties = configText ? configObject?.flatten() as Map<String, Object> : [:]
         Map writtenValues = filterMap.getWrittenValues()
         properties.putAll(writtenValues)
         new MapPropertySource(resource.filename, properties)
     }
 
     private NavigableMapPropertySource loadYamlConfig(Resource resource) {
         log.info("Loading YAML config file {}", resource.URI)
-        NavigableMapPropertySource propertySource = yamlPropertySourceLoader.load(resource.filename, resource, null) as NavigableMapPropertySource
+        NavigableMapPropertySource propertySource = yamlPropertySourceLoader.load(resource.filename, resource, null)?.first() as NavigableMapPropertySource
```

Comment:
We need to make sure that yml resources looking like this:
``` 
prop1:
   level1:
     value1: My value 1
----
prop2
   value2: My value 2
```
Does not mean, that we have two `NavigableMapPropertySource` entries

Reply:
Making a test now.